app/controllers/comments_controller.rb

```diff
@@ -16,4 +16,12 @@ def create
       render :new
     end
   end
+
+  def destroy
+    @user = @current_user
+    @post = Post.find_by(id: params[:post_id])
```

Comment:
- [ ]  You should use `load_and_authorize_resource`. It  is a helpful feature in Ruby on Rails that automatically loads and authorizes resources in a controller. It eliminates the need to manually load and authorize resources for each action, reducing code duplication. please use it in your post controller so that right method will be  consumed for authorized user.  for [more](https://github.com/CanCanCommunity/cancancan)